Fixed compilation under gcc 8.x

borkra opened the pull request:
This  resolves few issues preventing project compiling under GCC 8.x. 
The description/explanation of the solution below:

Revert commit 8e4f0ec38de04617c2433d0c038baef700b59805
and resolve the problem it was solving in a different manner
that does not result in kernel compilation failure under GCC 8

In addition iproute2 configure script is fixed and
uhttd compilation under glibc

If one would like to integrate this into local build he needs make sure that all compiled code is removed (make dirclean) before compiling this patch.

wongsyrone commented:
I think you should split this into several topic-based commits and rewrite commit title to make it clear.

borkra commented:
I am not sure what you propose, the  objective here is to fix compilation under GCC 8.x. 
The commit description describes the method of solving the problem - compilation failure.

There is more then 1 issue arising when compiling under gcc 8.x and these issues were fixed.

Splitting this up will result with project not compiling under gcc 8, if only 1 of the fixes applied.

borkra committed:
Fixed toolchain compilation for gcc 8.x

Revert commit 8e4f0ec and resolve the problem it was solving in a different
manner that does not result in kernel compilation failure under GCC 8

Signed-off-by: Boris Krasnovskiy <borkra@gmail.com>

borkra committed:
Fix uhttd compilation under glibc

Signed-off-by: Boris Krasnovskiy <borkra@gmail.com>

borkra committed:
Fixed iproute2 configure script under gcc 8.x

Signed-off-by: Boris Krasnovskiy <borkra@gmail.com>
